script/intentfest/conversation.py

```diff
@@ -0,0 +1,145 @@
+            if key not in fallbackLanguageSet:
+                raise Exception(
+                    f'Can\'t find the key "{key}" in language set "{self.language}" (and in default language set "{self.fallbackLanguage}" also).'
+                )
+            return fallbackLanguageSet[key]
+        return languageSet[key]
+
+
+class Conversation:
+    def __init__(self, language, debug=False) -> None:
+        self.language = language
+        self.localDebug = False if debug is None or debug is False else bool(int(debug))
+
+        self.sentenceParser = SentenceParser(self.language)
+        self.instructionsI18n = InstructionsI18n(self.language)
+
+        self.lastSentence = ""
+
+    def askUser(self):
+        promptSign = "<<<"
+        userSentence = input(f"{promptSign} ")
+
+        # Toggle debug
+        if userSentence.lower() == "d":
+            self.localDebug = not self.localDebug
+            print(
+                f"{self.instructionsI18n.get('DEBUG')}: {self.instructionsI18n.get('DEBUG_ON') if self.localDebug else self.instructionsI18n.get('DEBUG_OFF')}"
+            )
+
+        # Do intents reload
+        elif userSentence.lower() == "r":
+            self.sentenceParser.prepareNewIntents(self.language)
+            print(self.instructionsI18n.get("NEW_INTENTS_RELOADED"))
+
+        # Repeat the last query
+        elif userSentence.lower() == "a" or userSentence == "\u001b[A":
+            userSentence = self.lastSentence
+            print(f"{promptSign} {userSentence}")
+            return userSentence
+        else:
+            return userSentence
+
+        return False
+
+    def do(self):
+        print(self.instructionsI18n.get("HELLO_TEXT"), "\n")
+
+        while True:
+            userSentence = self.askUser()
+            if userSentence is not False:
+                self.lastSentence = userSentence
+                responseForUser = None
+                responseDebug = None
+
+                try:
+                    if self.sentenceParser.parse(userSentence):
+                        responseForUser = self.sentenceParser.getResponseForHuman()
+                        responseDebug = self.sentenceParser.getWholeResponseDebug()
+                except Exception:
+                    print(self.instructionsI18n.get("ERROR_OCCURRED"))
+                    traceback.print_exc()
+
+                print(
+                    f'>>> {responseForUser if responseForUser is not None else self.instructionsI18n.get("NO_AVAILABLE_RESPONSE")}'
+                )
+                if self.localDebug:
+                    print(f'{self.instructionsI18n.get("DEBUG")}: {responseDebug}')
+                    print("------------------------------")
+
+
+def run() -> int:
+    args = get_arguments()
+    conversation = Conversation(args.language, args.debug_response)
+    conversation.do()
+    # Liter said 'Missing return statement',
+    # but this particular return is unreachable.
+    # That is why it is here.
```

Comment:
If it doesn't come here, you didn't add a mode to quit the conversation :)

Reply:
Now we can end the conversation by typing "q" or pressing CTRL+C 😉